include/vast/Interfaces/AST/TypeInterface.td

```diff
@@ -6,6 +6,37 @@
 include "mlir/IR/OpBase.td"
 include "vast/Interfaces/AST/Common.td"
 
+def VAST_TypeInterface : VAST_ASTTypeInterface< "TypeInterface" > {
 
+    let description = [{
+        An interface to provide [`clang::Type`]
+        (https://clang.llvm.org/doxygen/classclang_1_1Type.html) functionality.
+    }];
+
+    let methods = [
+        InterfaceMethod< "clang::Type::isScalarType", "bool", "isScalarType", (ins) >,
+        InterfaceMethod< "clang::Type::isVectorType", "bool", "isVectorType", (ins) >,
+        InterfaceMethod< "clang::Type::isRecordType", "bool", "isRecordType", (ins) >,
+        InterfaceMethod< "clang::Type::isVoidType", "bool", "isVoidType", (ins) >,
+        InterfaceMethod< "clang::Type::isRVVSizelessBuiltinType", "bool", "isRVVSizelessBuiltinType", (ins) >,
+    ];
+}
+
+def VAST_QualTypeInterface : VAST_ASTTypeInterface< "QualTypeInterface",
+    [VAST_TypeInterface] > {
```

Comment:
Is the inheritance here correct? `clang::QualType` does not inherit from `clang::Type`, although it holds it via a pointer union.